src/snowflake/snowpark/mock/_connection.py

```diff
@@ -136,15 +136,34 @@ def write_table(
                 # Fix append by index
                 if name in self.table_registry:
                     target_table = self.table_registry[name]
+                    input_schema = table.columns.to_list()
+                    existing_schema = target_table.columns.to_list()
 
-                    if len(table.columns.to_list()) != len(
-                        target_table.columns.to_list()
+                    if len(input_schema) <= len(existing_schema) and (
+                        all(
+                            target_table[col].sf_type.nullable
+                            for col in (existing_schema[len(input_schema) :])
+                        )
                     ):
+                        for col in existing_schema[len(input_schema) :]:
+                            table[col] = None
+                            table.sf_types[col] = target_table[col].sf_type
+                    elif set(input_schema) != set(
```

Comment:
@sfc-gh-stan This particular condition doesn't seem to be working when the column set is different. Any chance you know what's going on here?